packages/test/test-end-to-end-tests/src/test/stashedOps.spec.ts

```diff
@@ -868,6 +868,71 @@ describeNoCompat("stashed ops", (getTestObjectProvider) => {
 		await waitForContainerConnection(container2);
 	});
 
+	it("resends DDS attach op", async function () {
+		const newMapId = "newMap";
+		const pendingOps = await getPendingOps(provider, false, async (_, dataStore) => {
+			const channel = dataStore.runtime.createChannel(
+				newMapId,
+				"https://graph.microsoft.com/types/map",
+			);
+			assert.strictEqual(channel.handle.isAttached, false, "Channel should be detached");
+
+			((await channel.handle.get()) as SharedObject).bindToContext();
+			assert.strictEqual(channel.handle.isAttached, true, "Channel should be attached");
+			(channel as SharedMap).set(testKey, testValue);
+		});
+
+		const container2 = await loader.resolve({ url }, pendingOps);
+		await waitForContainerConnection(container2);
+
+		// get new DDS from first container
+		const dataStore1 = await requestFluidObject<ITestFluidObject>(container1, "default");
+		const map2 = await requestFluidObject<SharedMap>(dataStore1.runtime, newMapId);
+		await provider.ensureSynchronized();
+		assert.strictEqual(map2.get(testKey), testValue);
+	});
+
+	it("handles stashed ops for local DDS", async function () {
+		const newCounterId = "newCounter";
+		const container = (await provider.loadTestContainer(
+			testContainerConfig,
+		)) as IContainerExperimental;
+		const defaultDataStore = await requestFluidObject<ITestFluidObject>(container, "/");
+
+		await provider.opProcessingController.pauseProcessing(container);
+
+		// create new DDS
+		const channel = defaultDataStore.runtime.createChannel(
+			newCounterId,
+			"https://graph.microsoft.com/types/counter",
+		);
+		assert.strictEqual(channel.handle.isAttached, false, "Channel should be detached");
+		((await channel.handle.get()) as SharedObject).bindToContext();
+		assert.strictEqual(channel.handle.isAttached, true, "Channel should be attached");
+
+		// op referencing new DDS is submitted at some later time (not in the same JS turn, so not batched)
+		await Promise.resolve();
+		(channel as SharedCounter).increment(0);
+		const stashP = new Promise<string>((resolve) => {
+			container.on("op", (op) => {
+				// Stash right after we see the DDS attach op. If we stash the DDS attach op, it will be applied
+				// first and everything will work fine. If ops are arriving on the network, there's no guarantee
+				// of how small this window is.
+				if (JSON.stringify(op).includes("attach")) {
+					resolve(container.closeAndGetPendingLocalState());
+				}
+			});
+		});
+		provider.opProcessingController.resumeProcessing(container);
+		const stashedOps = await stashP;
+
+		// when this container tries to apply the stashed DDS op, it will not have replayed the DDS attach
```

Comment:
is this a bug is how we re-arrange ops to put attaches first? specifically, does the reference sequence number of the attach op need to be lower than that of all other ops that have not been sent?

Reply:
I could be wrong, but I think only DataStore attach ops are sent out of order, not DDS attach ops

> specifically, does the reference sequence number of the attach op need to be lower than that of all other ops that have not been sent?

I don't think so. The reference sequence number of the attach op is the same as the DDS op, but its sequence number will be higher

Reply:
maybe #16579 is related

Reply:
I don't think it's related. The basic issue here is that we apply stashed ops at their reference sequence number because that's the state the document was in when it was created, but this doesn't account for local state like a pending DDS attach that is not captured when we stash. When the op was originally created, the client had a pending DDS, but the rehydrated client does not.

Reply:
ok. i understand the scenario, but i still don't quite understand how it happens, and why it didn't happen on other drivers, my guess is timing, but i'm not sure where timing comes into play.

I think we can mark these tests as skipped, and check them in as they are, as it looks like they hit the assert based on the PR run.

I assume the next step is to fix whatever the issue is?

Reply:
I would guess timing too. We don't really test this scenario in the stress test so it can only happen if ops from initialization are stashed. Maybe t9s is just really slow, or maybe it has something to do with the socket hangup errors I keep seeing.